mmeds/tools/tool.py

```diff
@@ -235,6 +237,100 @@ def summary(self):
         ]
         self.jobtext.append(' '.join(cmd))
 
+    def source_activate(self, env):
+        """ Change anaconda3 environment """
+        # Some possible envs: 'qiime', 'pheniqs', 'mmeds'
+        full_env = env
+        if env == 'qiime':
```

Comment:
if.... elif.... should have an else: raise ValueError() 
even if env can only take on two parameter values, if this is a function you'll build on and use in the future, it's more important to have. Especially since env doesn't have a default

Reply:
the if statement is just to allow for simpler names than required. So when I call `source_activate("qiime")` it knows to activate `qiime2-2020.8.0`. But I still want to be able to say `source_activate('lefse')` and have it activate the lefse environment. I suppose I could add elifs for each environment I use, but it would end up looking like `elif env == 'lefse': full_env = 'lefse'` which is rather redundant

Reply:
then there should be an else: `full_env=env` right?

I think redundant code is ok if it makes for clearer code

Reply:
i see the comment now though on line 242, that does help to know what other values env might take on